### ISSUE TITLE
Gallery block: add ability to set gutter

### ISSUE THREAD

**Issue by glendaviesnz** [WordPress/gutenberg#33582]
Once the gallery refactor is merged it would be good to add an option to allow the setting of the gutter between images to allow the gape between images to be expanded or removed.

**Comment by paaljoachim** [comment]
If I remember correctly I believe @aaronrobertshaw is working on the general gutter control which when merged can be added to multiple blocks. I came across a few issues when I searched for the word gutter here on the Gutenberg repo.

**Comment by aaronrobertshaw** [comment]
Thanks for the ping @paaljoachim. 

We did originally have a feature in the refactored gallery to [control gutter size](https://github.com/WordPress/gutenberg/pull/28377) however it was requested that be moved to a [future PR](https://github.com/WordPress/gutenberg/pull/29216). I also had a [PR](https://github.com/WordPress/gutenberg/pull/31987) to control gutters for columns via margin block support however that relied on CSS variables which were deemed to be undesirable.

Since then a promising [new PR adding grid gap block support](https://github.com/WordPress/gutenberg/pull/32571) has been created by @andrewserong. That might be the best bet to adding control over gallery gutters in the near future.

**Comment by dashkevych** [comment]
It would be great if such functionality was added to the Gallery block. In that case, we would be able to eliminate unnecessary space at the bottom of the gallery block. Also, it would simplify CSS of the Gallery block (no longer need to use `width: calc(25% - 0.75em);` and. `margin-right: 1em;`).

Each item in gallery adds side and bottom margins:

![Singular gallery item](https://snipboard.io/by9KVN.jpg)

As a result it creates additional space at the bottom of the Gallery block:

![Gallery block](https://snipboard.io/oTdr56.jpg)


Also, setting a default gap value for the Gallery block in `theme.json` would be very useful.

**Comment by glendaviesnz** [comment]
Done in #38164
